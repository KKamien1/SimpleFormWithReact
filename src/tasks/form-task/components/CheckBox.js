
import React from 'react';

export default function CheckBox({
    field: { name, value, required },
    children,
    options = [true, false],
    checked,
    onChange }) {

    const [yes, no] = options;

    const onChangeHandler = (e) => {
        e.target.value = e.target.checked ? yes : no;
        onChange(e)
    }
    return (
        <div className="form__input-field">
            <label htmlFor={name}>
                {children}
                {required ? <span className='form__asterisk'>*</span> : null}
                <input
                    name={name}
                    value={value}
                    onChange={onChangeHandler}
                    type="checkbox"
                    defaultChecked={checked}
                />
                <span>{yes}</span>
            </label>
        </div>
    )
}
