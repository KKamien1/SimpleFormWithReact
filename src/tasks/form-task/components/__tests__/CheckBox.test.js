import React from 'react';
import { shallow } from 'enzyme';

import CheckBox from '../CheckBox'

describe('CheckBox component', () => {
    test('should match to snapshot', () => {
        const component = shallow(<CheckBox />);
        expect(component.html()).toMatchSnapshot();
    })
    test('should display asterisk when field is required', () => {
        const component = shallow(<CheckBox required />);
        expect(component.find('.form__asterisk').length).toEqual(1);
    })
})
