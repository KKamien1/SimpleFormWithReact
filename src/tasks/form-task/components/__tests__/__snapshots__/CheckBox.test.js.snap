// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component should match to snapshot 1`] = `"<div class=\\"form__input-field\\"><label for=\\"usertype\\">User type<input type=\\"checkbox\\" name=\\"usertype\\" value=\\"\\"/><span></span></label></div>"`;
